bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,512 @@
+        error_info: Tuple[commands.Context, bool, str, Dict[str, int],
+                          datetime.time],
+        limiter: int = rhyming_line_finder_limiter
+    ) -> str:
+        """
+        Returns a sentence string with a last word in `word_rhymes`.
+
+        The function will continue to iterate through sentences provided by the
+        markov model, until it finds one that rhymes or until it reaches the
+        limiter.
+
+        The function also prevents the same line from appearing twice.
+        """
+        curr = 0
+
+        line = await self._get_sentence()
+        while (self._get_last_word(line) not in word_rhymes
+               and line not in existing_lines):
+            if curr >= limiter:
+                await self._handle_rhyming_sentence_not_found(*error_info)
+
+            line = await self._get_sentence()
+            curr += 1
+        return line
+
+    async def _init_unit(
+        self,
+        rhyme_track: Dict[str, Set[str]],
+        unit: str,
+        is_last_line: bool
+    ) -> str:
+        """
+        Returns a line with a last word that has a rhyme set.
+
+        If there is only one line in this unit, then the last word does not
+        need a rhyme set.
+        """
+        rhyme_set = set()
+
+        async def _get_line_and_rhyme_set() -> Tuple[str, Set[str]]:
+            line = await self._get_sentence()
+
+            rhyme_set = await self._get_rhyme_set(
+                instance=self,
+                word=self._get_last_word(line)
+            )
+
+            return line, rhyme_set
+
+        if is_last_line:
+            line, rhyme_set = await _get_line_and_rhyme_set()
+        else:
+            while len(rhyme_set) == 0:
+                line, rhyme_set = await _get_line_and_rhyme_set()
+
+        rhyme_track[unit] = rhyme_set
+        return line
+
+    @command_wrapper
+    async def _get_generated_markov_poem(
+        self,
+        ctx: commands.Context,
+        scheme: str,
+        unit_count: Dict[str, int],
+        time_start: datetime.time,
+        is_first_error: bool
+    ) -> List[str]:
+        """
+        Generates a love poem with the markov chain and sends it off.
+
+        If the count of a unit is one, that means it is the last unit in the
+        rhyme scheme being processed.
+
+        The function will iterate through the units of scheme. If it is a new
+        unit, it will be added to `rhyme_track` as a key with the rhyme set
+        as the value. If a unit is found again, then then the last word of the
+        new line's rhyme set will be added to `rhyme_track`. This is so that
+        all the rhymes are not contingent on the first unit of the scheme.
+        """
+        lines = []
+        rhyme_track = {}  # Maps units to their accumulative rhyme sets
+
+        for unit in scheme:
+            # `last_line` means last line of a rhyme group
+            is_last_line = True if unit_count[unit] == 1 else False
```

Comment:
We can simplify this.
```suggestion
            is_last_line = unit_count[unit] == 1
```